src/components/functions.js

```diff
@@ -1,10 +1,167 @@
-import axios from 'axios';
-import {BASE_URL,OPTIMIZE_SVG} from '../constants/constants';
-
-export default  function optimizeSvg(dataUrl){
-    return axios.post(`${BASE_URL}${OPTIMIZE_SVG}`, { dataUrl })
-        .then(res => {
-            return res.data;
-        })
-        .catch(err => console.log(err));
+import axios from "axios";
+import React from "react";
+import DisplaySvg from "../components/displaySvg/displaySvg";
+import finalSvgDisplay from "../components/displaySvg/finalSvgDisplay.scss";
+import { BASE_URL, OPTIMIZE_SVG } from "../constants/constants";
+import base64 from "base-64";
+
+export const optimizeSvg = (dataUrl, svgo) => {
+  return axios
+    .post(`${BASE_URL}${OPTIMIZE_SVG}`, { dataUrl, svgo })
+    .then(res => {
+      return res.data.urlData;
+    })
+    .catch(err => console.log(err));
+};
+
+export const immutablePush=(arr, newEntry)=>{
+  return [].concat(arr, newEntry)
 }
+
+export const changeObj = (value, svgoObjectPlugins) => {
+  let newSvgoObject = svgoObjectPlugins.filter(option => {
```

Comment:
map would be better suited for this functionality

Reply:
replaced that with map and works fine still.